Add IL2001-IL2060 trimmer warnings (#25842)
dotnet/docs: docs/core/deploying/trim-warnings/il2056.md

---
title: "IL2056: A 'System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute' annotation on a property conflicts with the same attribute on its backing field"
description: "Learn about trim warning IL2056: DynamicallyAccessedMembersPropertyConflictsBackingField"
ms.date: 07/23/2021
ms.topic: reference
author: mateoatr
ms.author: matorre
f1_keywords:
  - "IL2056"
---

> # IL2055: Call to 'System.Type.MakeGenericType' cannot be statically analyzed by the trimmer

Reviewer: This was copy pasted from IL2055 - needs to be changed to "Trim analysis: 'DynamicallyAccessedMemberAttribute' on property 'property' conflicts with the same attribute on its backing field 'field'."

## Cause

> Property annotated with <xref:System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute> has same annotation as its backing field.

Reviewer: ```suggestion
Property annotated with <xref:System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute> also has that attribute on its backing field.
```

(The annotation can actually be different, it's just that the same attribute is used - in fact the warning is here specifically for the case where the annotation would be actually different, as linker will ignore one of them in such case).
